src/main/java/de/tum/in/www1/exerciseapp/domain/QuizSubmission.java

```diff
@@ -49,7 +64,35 @@ public QuizSubmission removeSubmittedAnswers(SubmittedAnswer submittedAnswer) {
     public void setSubmittedAnswers(Set<SubmittedAnswer> submittedAnswers) {
         this.submittedAnswers = submittedAnswers;
     }
-    // jhipster-needle-entity-add-getters-setters - Jhipster will add getters and setters here, do not remove
+
+    /**
+     * calculates the scores for this submission and all its submitted answers and saves them in scoreInPoints
+     * @param quizExercise the quiz this submission belongs to (is needed to have values for isCorrect in answer options)
+     */
+    public void calculateAndUpdateScores(QuizExercise quizExercise) {
+        // set scores for all questions
+        for (Question question : quizExercise.getQuestions()) {
+            // search for submitted answer for this question
+            for (SubmittedAnswer submittedAnswer : getSubmittedAnswers()) {
+                if (question.getId().longValue() == submittedAnswer.getQuestion().getId().longValue()) {
```

Comment:
this line appears several times in your code, can you refactor it into a proper method?

Reply:
one option would also be to implement the equals operator in the Question class, then you could also write question.equals(submittedAnswer.getQuestion())